### ISSUE TITLE
Link list not fully responsive (goes off-screen on narrow screens)

### ISSUE THREAD

**Issue by shanereichart** [tinymce/tinymce#6335]
When the Link list within the "Insert/Edit link" modal has nested submenus (class "tox-tiered-menu"), and these submenus are viewed on narrow screens, such as mobile devices, the submenu responsiveness doesn't behave as expected.

The nested tiers of the menu attempt to position themselves to right of the primary tier when there is sufficient space. When there is insufficient space, they position to the left. However, when the left side also lacks sufficient space, they are displayed (unusably) outside the side side of view pane as such:

![image](https://user-images.githubusercontent.com/16807422/101994682-2515a780-3c92-11eb-8021-f236a078604c.png)

The expected behavior is for it to snap to the right edge of the screen, and thus partially overlap the parent tier. This will require javascript to determine if the left side of the parent drop-down has insufficient screen space, then `right: 0;` CSS would be applied.

This affects the implementation of the nested list list in TinyMCE v5.x.

**Comment by lostkeys** [comment]
Yep, I can confirm this being a bug. We'll post here when we have a fix. Thanks for reporting this!

**Comment by shanereichart** [comment]
This issue was referenced in #6353 and is possibly related to it. Of note there are:

1. This issue might occur only when a `fixed_toolbar_container` element has `position: fixed;` styling.
2. When the 2nd tier of the nested submenus are rendered outside the left viewport, any 3rd-tier submenus are displayed to the right of that, thereby returning back into the viewport, as such:

![InkedTiny-MCE-link-list-issue_LI](https://user-images.githubusercontent.com/16807422/102380761-4d551d00-3f96-11eb-8042-1b87b52d3867.jpg)

**Comment by mikkorantalainen** [comment]
Is there any known workaround?

**Comment by shanereichart** [comment]
> Is there any known workaround?

Here's ours:

```
/* workaround an issue with TinyMCE Link list (https://github.com/tinymce/tinymce/issues/6335) */
.tox-tiered-menu .tox-menu {
	width: fit-content;
	max-width: min(80%, 450px);
}

.tox-tiered-menu .tox-menu .tox-collection__item-label { word-break: normal; }

@media screen and (max-width: 890px) {
    .tox-tiered-menu > .tox-collection--list:not(:first-child) {
		left: auto !important;
		right: 0 !important;
	}
}
```
